### ISSUE TITLE
First completed filename doesn't get [path] highlighting

### ISSUE THREAD

**Issue by dawidd6** [zsh-users/zsh-syntax-highlighting#632]
```sh
# .zshrc
export ZSH="$HOME/.oh-my-zsh"
export HOMEBREW_PREFIX="/home/linuxbrew/.linuxbrew"
export PATH="$HOME/go/bin:$HOMEBREW_PREFIX/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games"

ZSH_THEME="robbyrussell"

if ! test -d "$ZSH"; then
    git clone --depth=1 https://github.com/robbyrussell/oh-my-zsh "$ZSH"
fi

if type brew &>/dev/null; then
    eval $(SHELL=zsh brew shellenv)
fi

source $ZSH/oh-my-zsh.sh
source $HOMEBREW_PREFIX/share/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh
```

When one for example writes `docker-comp`, completes it with TAB and then presses TAB to get a completion inserted, it is inserted, but in bold. This does not happen when `oh-my-zsh` is not sourced or command is entered manually from start to the end.

[![asciicast](https://asciinema.org/a/WOJ4ZWvCyemxXD6yuyYAh5zGl.svg)](https://asciinema.org/a/WOJ4ZWvCyemxXD6yuyYAh5zGl)

As one can see, entering TAB right after command, produces bold completion and when a space is already in place after command the inserted text is not bold as it should be.

**Comment by z0rc** [comment]
Bold text representation depends on terminal emulator. Look into terminal settings for options like "Draw intense colors with bold font". If this true for you, adjust highlight styles to better match your preference and terminal color palette.

**Comment by danielshahaf** [comment]
> This does not happen when oh-my-zsh is not sourced

Please provide a minimal reproducer using `zsh -f` with no external dependencies. 

Please confirm that disabling zsh-syntax-highlighting fixes the symptoms.

**Comment by danielshahaf** [comment]
I've been able to reproduce something similar to this.

With z-sy-h master and zsh master (I haven't checked earlier versions of either), if I do:

```zsh
$ zsh -f
% source ./zsh-syntax-highlighting.zsh
% ls <TAB><TAB>…
```

then after the first `<TAB>`, the filename shown isn't underlined.  After subsequent `<TAB>` presses the filenames get underlined.  I press `<TAB>` enough times for the first filename to be offered a second time and at that point it did get underlined.

@dawidd6 When I requested a reproducer, this is what I meant: a minimal sequence of steps that starts from `zsh -f` and demonstrates the problem.
